pkg/network/usm/monitor_tls_test.go

```diff
@@ -507,6 +507,23 @@ func (s *tlsSuite) TestJavaInjection() {
 	}
 }
 
+func TestGoTLSInitialization(t *testing.T) {
+	// This is meant to test the eBPF program loading in isolation to detect
+	// things such as verifier errors
+	modes := []ebpftest.BuildMode{ebpftest.RuntimeCompiled, ebpftest.CORE}
+	ebpftest.TestBuildModes(t, modes, "", func(t *testing.T) {
+		cfg := config.New()
+		cfg.EnableGoTLSSupport = true
+
+		if !gotlstestutil.GoTLSSupported(t, cfg) {
+			t.Skip("GoTLS not supported for this setup")
+		}
+
+		monitor := setupUSMTLSMonitor(t, cfg)
+		assert.NotNil(t, monitor)
+	})
+}
```

Comment:
We already have `TestFullMonitorWithTracer`
why didn't it capture the issue?

Reply:
I think we should simply need to add `require.NotNil(t, tr.usmMonitor)` to the test I shared